### PR TITLE
chore(rds): add Aurora MySQL version 2.10.1

### LINKED ISSUE #17449: [rds] Add latest version in AuroraMysqlEngineVersion
### Description

Update entries for AuroraMysqlEngineVersion to include the latest version available (2.10.1). 

```
aws rds describe-db-engine-versions --engine aurora-mysql | jq -r '.DBEngineVersions[] | [.Engine, .EngineVersion]'
...
...
...
[
  "aurora-mysql",
  "5.7.mysql_aurora.2.10.1"
]
```

### Use Case

Allow us to use the latest version of MySQL offered by the AWS RDS Aurora service.

### Proposed Solution

https://github.com/aws/aws-cdk/blob/6aec58d564d390b7600f469203fe0326b55e1672/packages/%40aws-cdk/aws-rds/lib/cluster-engine.ts#L335-L339

Add version 2.10.1 to the AuroraMysqlEngineVersion class.

### Other information

_No response_

### Acknowledge

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### PR DESCRIPTION
**Pull request by cokia**
This is an addition to support the latest version of `5.7.mysql_aurora2.10.1` in `AuroraMysqlEngineVersion` in the CDK RDS package.

Closes #17449

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-rds/test/cluster-engine.test.ts [hunk 1]
```diff
@@ -59,7 +59,7 @@ describe('cluster engine', () => {
   test('cluster parameter group correctly determined for AURORA_MYSQL and given version', () => {
     // GIVEN
     const engine = DatabaseClusterEngine.auroraMysql({
-      version: AuroraMysqlEngineVersion.VER_2_07_1,
+      version: AuroraMysqlEngineVersion.VER_2_10_1,
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/test/cluster-engine.test.ts:62]: Let's leave the test as-is 🙂.

**Review by skinny85 — CHANGES_REQUESTED** [review]
Thanks for the contribution @cokia! Let's revert the change in the test, and we'll get this merged in.

File: packages/@aws-cdk/aws-rds/test/cluster-engine.test.ts (hunk 1 above)
Comment (cokia) [reply packages/@aws-cdk/aws-rds/test/cluster-engine.test.ts:62]: Is there a reason why I can not change the test for this one?

**Review by skinny85 — APPROVED** [review]
Thanks for the contribution @cokia!

File: packages/@aws-cdk/aws-rds/test/cluster-engine.test.ts (hunk 1 above)
Comment (skinny85) [reply packages/@aws-cdk/aws-rds/test/cluster-engine.test.ts]: We generally don't touch existing tests unless there's a pressing need to change them for some reason, and I don't see anything like that here 🙂.

**Pull request merged**